docs/serving/samples/machinelearning-python-bentoml/README.md

```diff
@@ -0,0 +1,175 @@
+# Python BentoML sample
+
+A simple machine learning model with API serving that is written in Python and
+using [BentoML](https://github.com/bentoml/BentoML).
+
+BentoML is an open source framework for high performance ML model serving, which supports all major machine learning frameworks including Keras, Tensorflow, PyTorch, Fast.ai, and XGBoost.
+
+This sample walks you through the steps of creating and deploying a machine learning model using Python. It uses BentoML to package a classifier model that is trained on the Iris dataset, and then creates a container image and
+deploy the image to Knative.
+
+Knative deployment guide with BentoML is also available in the [BentoML documentation](https://docs.bentoml.org/en/latest/deployment/knative.html).
+
+## Prerequisites
+
+- You must have a Kubernetes cluster with Knative installed. See the [installation instructions](../../../../docs/admin/install/README.md) for more information.
+- You must have [Docker](https://www.docker.com) installed and running on your local machine, and a Docker Hub account configured. Docker Hub is used as a container registry.
+- You must have Python 3.6 or above installed and running on your local machine.
+- You must install the `scikit-learn` and `bentoml` packages:
+
+    ```bash
+    pip install scikit-learn
+    pip install bentoml
+    ```
+
+## About the sample code
+
+### iris_classifier
+
+BentoML creates a model API server by using prediction service abstraction. In the `iris_classifier.py` file, a prediction service is defined, which:
+
+1. Requires a `scikit-learn` model.
+1. Asks BentoML to identify the required `pip` dependencies.
+1. Defines an API, which is the entry point for accessing this machine learning service.
+
+--8<-- "iris_classifier.py"
```

Comment:
@omerbensaadon @julz howwww do I include .py files? 😭 This just doesn't show up but I wasn't sure if `include` works for non-md things. We should figure this out anyway for the whole where-do-samples-go discussion

Reply:
ah, right, we [currently have the base path for snippets set to `docs/snippets`](https://github.com/knative/docs/blob/mkdocs/mkdocs.yml#L36), so this would have to be:

~~~suggestion
```python
--8<-- "../serving/samples/machinelearning-python/bentoml/iris_classifier.py"
```
~~~

Maybe we should set base path to `docs/` instead (and then the snippets would be linked as `--8<-- "snippets/blah.md"`, which might be clearer anyway 🤔). Unfortunately [it looks like](https://github.com/mkdocs/mkdocs/issues/2154) mkdocs doesn't yet support using current-file-relative links in snippets :(.

Reply:
+1 for setting the base path to `/docs`, otherwise it's going to make snippets really hard to organize because we'll have a bunch of random topics and files in one directory 🙂 
@julz can you open an issue to document this for contributing docs? I feel like this is something we should have written down so folks know how to handle it going forward.